src/editor-service.ts

```diff
@@ -63,14 +65,40 @@ export class WarningCarryingException extends Error {
   }
 }
 
-export class EditorService {
+// Important note: all arguments to, and results returned from editor service
+//     methods MUST be serializable as JSON, as the editor service may be
+//     running out of process and communicating with JSON strings.
+//
+//     Fortunately, editor-service_test will test that the results are JSON
+//     serializable.
+export abstract class BaseEditor {
+  abstract async fileChanged(localPath: string, contents?: string):
+      Promise<void>;
+
+  abstract async getDocumentationFor(localPath: string, position: Position):
+      Promise<string|undefined>;
+
+  abstract async getDefinitionFor(localPath: string, position: Position):
+      Promise<SourceRange>;
```

Comment:
what if there's no definition found? Should the return type be `Promise<SourceRange|undefined>`, or is that a rejection? This should be documented.